### ISSUE TITLE
Updating of assets

### ISSUE THREAD

**Issue by git-masoud** [eclipse-edc/Connector#2477]
# Feature Request
Updates an asset with the given ID via the Data Management Api. To update only the properties and data address of an asset. 

## Which Areas Would Be Affected?
asset-api
control-plane-core
control-plane-spi
asset-index-sql
asset-index-cosmos

## Why Is the Feature Desired?

EDC is not supporting updating of the assets, which can be limiting in certain scenarios. For instance, if there is a typo in the asset description, it is not possible to correct it without creating a new asset, and an asset can only be deleted if it is not referenced in a contract agreement. Users should be able to update the asset's properties or its data address without deleting the whole asset.

## Solution Proposal
This feature should be similar to the asset creation process, with the only difference being that only the properties and data address of the asset should be removed first and created with the new information.

**Comment by paullatzelsperger** [comment]
@git-masoud the fact that assets _cannot be updated_ was by design and on purpose for various reasons. Changing that would require a decision by the EDC technical committers, which - to my knowledge - has not happened. This issue and the associated PR is premature and I personally find it quite frankly a bit presumptuous.

The correct way to approach this would be to create a discussion first and let committers decide on the merits of the feature, and if asked, provide a decision record and technical outline.

**Comment by git-masoud** [comment]
Thank you for your response and explanation @paullatzelsperger . 
now, I understand that changing this feature would require a decision by the EDC committers. However, I reached out to @ndr-brt to get further guidance on how to proceed on this feature request and he suggested that I create a decision record to document the discussion and decision-making process. I apologize for any misunderstanding or confusion my initial request may have caused.

I wanted to clarify that my request for the asset update feature was based on a clear need to update certain information in the assets, beyond just correcting typos. As we continue to use EDC, we have found that there are some pieces of information that need to be updated periodically, and it would be more efficient to update them directly rather than having to delete and recreate the asset. Furthermore, it is not possible to delete an asset if it is referenced in a contract agreement.

With that said, I appreciate your suggestion to start a discussion and will raise a discussion about it soon.
